modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.js

```diff
@@ -270,12 +271,48 @@ const ActionsCell = ({item, itemId, itemsActions}) => {
 					actions={itemsActions || item.actionDropdownItems}
 					itemData={item}
 					itemId={itemId}
+					menuActive={menuActive}
+					onMenuActiveChange={onMenuActiveChange}
 				/>
 			)}
 		</DndTable.Cell>
 	);
 };
 
+const RowWithActions = ({
+	active,
+	children,
+	className,
+	item,
+	itemId,
+	itemsActions,
+	selected,
+	...props
+}) => {
+	const [menuActive, setMenuActive] = useState(false);
+
+	return (
+		<DndTable.Row
+			className={classNames(className, {
+				active,
+				expanded: menuActive,
+				selected,
+			})}
+			{...props}
+		>
+			{children}
+
+			<ActionsCell
+				item={item}
+				itemId={itemId}
+				itemsActions={itemsActions}
+				menuActive={menuActive}
+				onMenuActiveChange={setMenuActive}
+			/>
```

Comment:
We can move `ActionCell` here inline. The only reason I moved `ActionCell` outside of main component is to reduce high amount of line breakage on deep indentation.

For this purpose, we could even move all of `{children}` here inline. It'll add some boilerpate with props, but it's probably worth it to make the code more readable.

Reply:
Done! Moved `ActionCell` to inline and also all `{children}` elements. [Also changed `expanded` to `menu-active`](https://github.com/liferay-frontend/liferay-portal/pull/2866#discussion_r1030377652). See https://github.com/liferay-frontend/liferay-portal/pull/2866/commits/cc62e541f519255d3d41d5d20a403d21689765d1